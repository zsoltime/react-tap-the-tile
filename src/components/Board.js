import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/macro';

import Tile from '../containers/Tile';

const StyledBoard = styled.div`
  align-items: center;
  display: flex;
  flex-wrap: wrap;
  height: 20rem;
  justify-content: space-around;
  margin-bottom: 1rem;
  width: 20rem;

  @media (min-width: 420px) {
    height: 25rem;
    width: 25rem;
  }
`;

const Board = ({ tiles, onClick, onTimeout }) => (
  <StyledBoard>
    {Object.keys(tiles).map(i => {
      const tile = tiles[i];

      return (
        <Tile
          key={i}
          clicksRemaining={tile.clicksRemaining}
          id={i}
          isActive={tile.isActive}
          onClick={onClick}
          onTimeout={onTimeout}
        />
      );
    })}
  </StyledBoard>
);

Board.propTypes = {
  onClick: PropTypes.func.isRequired,
  onTimeout: PropTypes.func.isRequired,
  tiles: PropTypes.object.isRequired,
};

export default Board;
