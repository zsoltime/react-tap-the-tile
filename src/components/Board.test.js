import React from 'react';
import { mount, render } from 'enzyme';
import 'jest-styled-components';

import Board from './Board';

describe('<Board />', () => {
  it('renders a list of buttons', () => {
    const tiles = [
      {
        clicksRemaining: 3,
        isActive: true,
      },
      {
        clicksRemaining: 0,
        isActive: false,
      },
      {
        clicksRemaining: 1,
        isActive: true,
      },
    ];
    const wrapper = render(
      <Board tiles={tiles} onClick={f => f} onTimeout={f => f} />
    );

    expect(wrapper.find('button').length).toBe(tiles.length);
    expect(wrapper).toMatchSnapshot();
  });

  it('passes onClick prop to <Tile />', () => {
    const tiles = [
      {
        clicksRemaining: 1,
        isActive: true,
      },
      {
        clicksRemaining: 0,
        isActive: false,
      },
    ];
    const handleClick = jest.fn();
    const buttonId = 0;
    const timeRemaining = 5000;
    const wrapper = mount(
      <Board tiles={tiles} onClick={handleClick} onTimeout={f => f} />
    );

    wrapper
      .find('button')
      .first()
      .simulate('click');

    expect(handleClick).toBeCalledTimes(1);
    expect(handleClick).toBeCalledWith(buttonId, timeRemaining);
  });
});
