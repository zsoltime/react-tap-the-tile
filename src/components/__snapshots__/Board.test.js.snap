// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> renders a list of buttons 1`] = `
.c1 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  background-color: #363636;
  border: 1px solid rgba(0,0,0,0.8);
  color: rgba(255,255,255,0.95);
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 23%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-transition: background 0.3s ease-in-out;
  transition: background 0.3s ease-in-out;
  width: 23%;
}

.c4 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  background-color: #fff;
  border: 1px solid rgba(0,0,0,0.8);
  color: rgba(255,255,255,0.95);
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 23%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-transition: background 0.3s ease-in-out;
  transition: background 0.3s ease-in-out;
  width: 23%;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 0.825rem;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  height: 20rem;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  margin-bottom: 1rem;
  width: 20rem;
}

@media (min-width:420px) {
  .c0 {
    height: 25rem;
    width: 25rem;
  }
}

<div
  class="c0"
>
  <button
    class="c1"
    type="button"
  >
    <div
      class="c2"
    >
      3
    </div>
    <div
      class="c3"
    >
      5
    </div>
  </button>
  <button
    class="c4"
    type="button"
  />
  <button
    class="c1"
    type="button"
  >
    <div
      class="c2"
    />
    <div
      class="c3"
    >
      5
    </div>
  </button>
</div>
`;
